common/ManagementTestShared/Redesign/ManagementRecordedTestBase.cs

```diff
@@ -202,11 +205,11 @@ public void OneTimeCleanupResourceGroups()
                 Parallel.ForEach(OneTimeResourceGroupCleanupPolicy.ResourceGroupsCreated, resourceGroup =>
                 {
                     var sub = _cleanupClient.GetSubscriptions().GetIfExists(SessionEnvironment.SubscriptionId);
-                    sub.Value?.GetResourceGroups().Get(resourceGroup).Value.Delete(waitForCompletion: false);
+                    sub.Value?.GetResourceGroups().Get(resourceGroup).Value.Delete(waitForCompletion: _waitForCleanup);
                 });
                 Parallel.ForEach(OneTimeManagementGroupCleanupPolicy.ManagementGroupsCreated, mgmtGroupId =>
                 {
-                    _cleanupClient.GetManagementGroup(mgmtGroupId).Delete(waitForCompletion: false);
+                    _cleanupClient.GetManagementGroup(mgmtGroupId).Delete(waitForCompletion: _waitForCleanup);
```

Comment:
is the idea here to make it sequential and blocking? If so then there's `Parallel.Foreach` wrapping this (and other places like this). Should that also change?

Reply:
the idea is to make the test case wait for the clean up to happen when in live mode.  In playback / recording this isn't necessary and adds 5-10 minutes to each test case.  The reason this is needed is the seed value does get shared in some cases for generating the random asset name and in those cases one test case is deleting the resource and another test case is trying to create one with the same name.

The parallel foreach operates around all the resourcegroups or mgmt groups that were created for the test cases since some of them do create multiple however in most cases this will be 1